Engine/ScriptAnalyzer.cs

```diff
@@ -1475,6 +1477,13 @@ public IEnumerable<DiagnosticRecord> AnalyzePath(string path, bool searchRecursi
             this.BuildScriptPathList(path, searchRecursively, scriptFilePaths);
             foreach (string scriptFilePath in scriptFilePaths)
             {
+                if (autoFix)
+                {
```

Comment:
I think the right place for this block is in the InvokeScriptAnalyzerCommand.cs file. We do not the behavior 'Analyzepath` method, which should only analyze and not fix.

Reply:
This would require some re-factoring of the project because one needs to call the private method `BuildScriptPathList` of the `ScriptAnalyzer` class first to get the list of files, therefore it would probably be best to extract the `BuildScriptPathList` method into a separate class. Can you confirm that you really want this or suggest other options?